lib/nrf_modem_lib/nrf91_connectivity/nrf91_connectivity.h

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF91_CONNECTIVITY_H__
+#define NRF91_CONNECTIVITY_H__
+
+#include <zephyr/types.h>
+#include <zephyr/net/conn_mgr_connectivity.h>
+
+/**
+ * @defgroup nrf91_connectivity nRF91 connectivity layer
+ * @{
+ * @brief Library that provides connectivity for nRF91 based boards.
+ *	  The library is intended to be integrated with and bound to a Zephyr network interface.
+ *	  It can be managed via Zephyr's network management API and the
+ *	  Connection Manager Connectivity API using calls such as net_if_down(), net_if_up(), and
+ *	  conn_mgr_if_connect().
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Options used to specify desired behavior when the network interface is brough down by
+ *	   calling net_if_down().
+ */
+enum nrf91_connectivity_if_down_options {
+	/** Shutdown modem - GNSS functionality will not be preserved. */
+	NRF91_CONNECTIVITY_IF_DOWN_MODEM_SHUTDOWN = 1,
+
+	/** Disconnect from LTE - GNSS functionality will be preserved. */
+	NRF91_CONNECTIVITY_IF_DOWN_LTE_DISCONNECT
+};
+
+/** @brief Option types used to specify library behavior. */
+enum nrf91_connectivity_options {
+	/** Sets desired behavior when nrf91_connectivity_disable() / net_if_down() is called. */
+	NRF91_CONNECTIVITY_IF_DOWN = 1,
+};
```

Comment:
In practise I don't expect users to set these options more than once. But, if there is a good use case / demand for it I'm happy to add it. I just want to prevent having features that won't be used.

Reply:
Yeah, perhaps runtime setting of PSM/eDRX is too unlikely a desire to be supported via conn_mgr_if_xxx_opt accessors, fair enough